### PR TITLE
Fixed showing unknown info from UI issue #628

### PR DESCRIPTION
**Pull request by MLPJ**
Fixed showing unknown info from UI #628 #651
Changes: 
fixed the issue in 3 views,

1. organizer_detail_layout.xml which shows organizer details
2. about_event_details.xml which shows event details
3. bottomsheet_attendee_check_in.xml which popup from bottom when an attendee is selected

Screenshots for the change: 
Here is a screenshot before the fix
screenshot_2018-02-20-03-12-38

Here are the screenshots after fix
screenshot_2018-02-20-03-41-12
screenshot_2018-02-20-03-41-23
screenshot_2018-02-20-03-41-36

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Please resolve conflicts

File: app/src/main/res/layout/organizer_detail_layout.xml [hunk 1]
```diff
@@ -67,7 +70,8 @@
                         android:drawablePadding="@dimen/spacing_small"
                         android:drawableLeft="@{ @drawable/ic_email }"
                         android:drawableStart="@{ @drawable/ic_email }"
-                        android:text='@{ user.email }' />
+                        android:text='@{ user.email }'
+                        android:visibility="@{user.email == null ? View.GONE : View.VISIBLE}"/>
```
Comment (Masquerade0097) [inline app/src/main/res/layout/organizer_detail_layout.xml]: Email is a mandatory field so no need for null check.

File: app/src/main/res/layout/login_fragment.xml [hunk 2]
```diff
@@ -39,60 +39,67 @@
...
                         android:hint="@string/email"
-                        android:text="@={ login.email }"
-                        android:inputType="textEmailAddress"
                         android:imeOptions="actionNext"
-                        app:validateType='@{"email"}' />
+                        android:inputType="textEmailAddress"
+
+                        app:validateType='@{"email"}'
+
+                        android:text="@={ login.email }"
```
Comment (Masquerade0097) [inline app/src/main/res/layout/login_fragment.xml]: Please remove the unnecesaary new lines and spacing.

**Comment by kasvith** [comment]
@MLPJ to resolve merge conflicts please use a merge tool, Android Studio supports by default for this.
Make sure you add `upstream` repo for `https://github.com/fossasia/open-event-orga-app.git` using 
`git remote add upstream https://github.com/fossasia/open-event-orga-app.git`

**Comment by iamareebjamal** [comment]
Please update PRs instead of closing them

**Comment by MLPJ** [comment]
sorry for the inconvinience, android studio said that I can't make another PR that is why I closed the older one. any solutions?

**Comment by MLPJ** [comment]
@iamareebjamal  is that right now? sorry for the inconvinience

**Comment by iamareebjamal** [comment]
You don't need to create a new PR, just push to the branch from where the PR is made and it will be updated

**Comment by iamareebjamal** [comment]
There are various changes requested which are not completed yet

**Comment by MLPJ** [comment]
I did the changes sugested by @Masquerade0097, if there are still other things please let me know, I'll recheck too.
Thanks for the help @iamareebjamal.

File: app/src/main/java/org/fossasia/openevent/app/module/event/about/AboutEventActivity.java [hunk 3]
```diff
@@ -20,6 +20,7 @@ protected void onCreate(Bundle savedInstanceState) {
         if (extras != null && extras.containsKey(EVENT_ID)) {
             long id = extras.getLong(EVENT_ID);
 
+
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/module/event/about/AboutEventActivity.java]: Unnecessary change

File: app/src/main/res/layout/about_event_detail.xml [hunk 4]
```diff
@@ -190,19 +190,23 @@
                         android:drawableLeft="@{ @drawable/ic_earth }"
                         android:drawablePadding="@dimen/spacing_small"
                         android:drawableStart="@{ @drawable/ic_earth }"
+                        android:text='@{ event.paymentCountry }'
+                        android:visibility="@{ (event.paymentCountry != null) ? View.VISIBLE : View.GONE }"/>
                         android:gravity="center_vertical"
-                        android:padding="@dimen/spacing_small"
-                        android:text='@{ event.paymentCountry }' />
+                        android:padding="@dimen/spacing_small" />
+
```
Comment (iamareebjamal) [inline app/src/main/res/layout/about_event_detail.xml]: Unnecessary changes. One line is needed to be added and several are reordered. This is undesirable as it increases entropy of the PR and we think there are a lot of changes which need to be reviewed. It also pollutes the git blame system where the original author is shadowed by the one rearranging the code. That's why it's a suggestion to only format files in a separate code format PR

File: app/src/main/res/layout/about_event_detail.xml [hunk 5]
```diff
@@ -190,19 +190,23 @@
...
                         android:drawableLeft="@{ @drawable/ic_coin }"
                         android:drawablePadding="@dimen/spacing_small"
                         android:drawableStart="@{ @drawable/ic_coin }"
+                        android:text='@{ event.paymentCurrency }'
+                        android:visibility="@{ (event.paymentCurrency != null) ? View.VISIBLE : View.GONE }"/>
                         android:gravity="center_vertical"
-                        android:padding="@dimen/spacing_small"
-                        android:text='@{ event.paymentCurrency }' />
+                        android:padding="@dimen/spacing_small" />
+
```
Comment (iamareebjamal) [inline app/src/main/res/layout/about_event_detail.xml]: Similarly here

File: app/src/main/res/layout/about_event_detail.xml [hunk 6]
```diff
@@ -312,9 +316,10 @@
                         android:drawableLeft="@{ @drawable/ic_info }"
                         android:drawablePadding="@dimen/spacing_small"
                         android:drawableStart="@{ @drawable/ic_info }"
+                        android:text='@{ event.description }'
+                        android:visibility="@{ (event.description != null) ? View.VISIBLE : View.GONE }"/>
                         android:gravity="center_vertical"
-                        android:padding="@dimen/spacing_small"
-                        android:text='@{ event.description }' />
+                        android:padding="@dimen/spacing_small" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/about_event_detail.xml]: Similarly here

File: app/src/main/res/layout/login_fragment.xml (hunk 2 above)
Comment (iamareebjamal) [reply app/src/main/res/layout/login_fragment.xml]: This was the change I was talking about not being still addressed

File: app/src/main/res/layout/login_fragment.xml [hunk 7]
```diff
@@ -39,60 +39,67 @@
...
-                        android:inputType="textPassword"
                         android:imeOptions="actionGo"
+                        android:inputType="textPassword"
+
+                        app:go="@{() -> btnLogin.performClick() }"
                         app:validateMinLength="@{6}"
-                        app:go="@{() -> btnLogin.performClick() }" />
+                        android:text="@={ login.password }"
+                        />
+
```
Comment (iamareebjamal) [inline app/src/main/res/layout/login_fragment.xml]: Both extra lines and unneeded changes

File: app/src/main/res/layout/login_fragment.xml [hunk 8]
```diff
@@ -39,60 +39,67 @@
...
                 </android.support.design.widget.TextInputLayout>
 
                 <include
                     android:id="@+id/url"
                     layout="@layout/auth_url_container"
-                    app:action="@{ () -> btnLogin.performClick() }"/>
+                    app:action="@{ () -> btnLogin.performClick() }" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/login_fragment.xml]: Unneeded changes

File: app/src/main/res/layout/login_fragment.xml [hunk 9]
```diff
@@ -39,60 +39,67 @@
...
                 <android.support.v7.widget.AppCompatButton
                     android:id="@+id/btnLogin"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:layout_gravity="center"
-                    app:backgroundTint="@color/color_accent"
                     android:text="@string/login"
-                    android:textColor="@android:color/white" />
+                    android:textColor="@android:color/white"
+                    app:backgroundTint="@color/color_accent" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/login_fragment.xml]: Unneeded changes

File: app/src/main/res/layout/login_fragment.xml [hunk 10]
```diff
@@ -39,60 +39,67 @@
...
                 <android.support.v7.widget.AppCompatButton
                     android:id="@+id/signUpLink"
+                    style="?attr/borderlessButtonStyle"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:layout_gravity="center_horizontal"
                     android:text="@string/sign_up_link"
-                    android:textColor="@color/color_accent"
-                    style="?attr/borderlessButtonStyle" />
+                    android:textColor="@color/color_accent" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/login_fragment.xml]: Unneeded changes

File: app/src/main/res/layout/login_fragment.xml [hunk 11]
```diff
@@ -39,60 +39,67 @@
...
                 <android.support.v7.widget.AppCompatButton
                     android:id="@+id/forgotPasswordLink"
+                    style="?attr/borderlessButtonStyle"
                     android:layout_width="wrap_content"
                     android:layout_height="wrap_content"
                     android:layout_gravity="center_horizontal"
                     android:text="@string/forgot_password_link"
-                    android:textColor="@color/color_accent"
-                    style="?attr/borderlessButtonStyle" />
+                    android:textColor="@color/color_accent" />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/login_fragment.xml]: Unneeded changes

File: app/src/main/res/layout/organizer_detail_layout.xml [hunk 12]
```diff
@@ -110,7 +113,8 @@
                         android:text="@string/contact"
                         android:background="#dedede"
                         android:textAppearance="@style/TextAppearance.AppCompat.Headline"
-                        android:textColor="?android:textColorSecondary" />
+                        android:textColor="?android:textColorSecondary"
+                        />
```
Comment (iamareebjamal) [inline app/src/main/res/layout/organizer_detail_layout.xml]: Unneeded changes

File: app/src/main/res/layout/organizer_detail_layout.xml [hunk 13]
```diff
@@ -130,7 +135,8 @@
                         android:drawablePadding="@dimen/spacing_small"
                         android:drawableLeft="@{ @drawable/ic_facebook }"
                         android:drawableStart="@{ @drawable/ic_facebook }"
-                        android:text='@{ user.facebookUrl }' />
+                        android:text='@{ user.facebookUrl != null ? user.facebookUrl :  "test" }'
```
Comment (iamareebjamal) [inline app/src/main/res/layout/organizer_detail_layout.xml]: We don't want to show test if facebook URL is null

**Comment by MLPJ** [comment]
@iamareebjamal i'll correct those as soon as possible. Thanks for pointing out those points

**Comment by MLPJ** [comment]
@iamareebjamal  @Masquerade0097  sorry for the inconvenience caused earlier, I think now it is upto the standards. Can you please review again. If there are any further changes required please let me know.
